src/integration_test.rs

```diff
@@ -213,7 +213,9 @@ fn amm_add_and_remove_liquidity() {
     let lp_token = Cw20Contract(Addr::unchecked(info.lp_token_address));
 
     // check initial balances
-    let owner_balance = cw20_token.balance(&router, owner.clone()).unwrap();
+    let owner_balance = cw20_token
+        .balance::<App, Addr, Empty>(&router, owner.clone())
```

Comment:
first two generics can be inferred. best to keep things generic as possible where we can. :)

```suggestion
        .balance::<_, _, Empty>(&router, owner.clone())
```

Reply:
fixed here : 1b42d76915f6cd3de62186b80b9a368ca836ee10